google/google-how-to/gce/set-hostname-using-cloudinit.rst

```diff
@@ -1,20 +1,29 @@
 Set hostname of GCE instances
-=================================
+=============================
 
-The hostname of GCE instances can be set using multiple methods. Apart from setting it manually, one option is to use a set-up tool like cloud-init, while Google's preferred method is to use its DHCP service. Both these options are considered here.
+The hostname of GCE instances can be set using multiple methods. Google's preferred method is to use its DHCP service, which requires you to choose a fully qualified DNS name (FQDN), e.g. something like test123.test.com. If you don't want an FQDN or if you want a specific format for consistent naming across clouds, you can use a set-up tool like cloud-init to set your hostname.
 
-.. _use_cloud-init:
+Both these methods are described here. Also, due to a recent hostname-related update on GCP, you might have to make some additional changes for GCE images that use Ubuntu 24.04 and later. These are explained at the end.
 
-Option 1: Use cloud-init
-------------------------
 
-You can set the required hostname by programming the cloud-init metadata. For this, use the `gcloud compute instances add-metadata`_ command:
+Using DHCP (Google's preferred method)
+--------------------------------------
+
+By default, Google's DHCP service sets the hostname to an automatically generated internal DNS name. 
+
+To set your own custom name, follow the instructions given in `Create a VM instance with a custom hostname`_. In this case, the DHCP service will additionally provide the custom name and will prioritise it to be the default hostname. However, as mentioned earlier, the custom name needs to be an FQDN and you also need to create a corresponding DNS record for it.
```

Comment:
Can you expand on "you also need to create a corresponding DNS record for it"?  You can set the hostname, even the FQDN, without a DNS record.  The only reason you need a DNS record is if you want to be able to access the VM via the hostname, which is true regardless of whether you're setting it via Google DHCP or cloud-init (or just the hostname command on your VM).

Reply:
Oh yeah that makes sense, then I think I'll just remove that additional phrase. (From a quick reading of the google doc, it had seemed like creating a DNS record was mandatory - "With custom hostnames, you still need to create a corresponding DNS record in the appropriate zone, for example, by using Cloud DNS.")

Reply:
Yeah, Google makes a big thing out of it, I think because they want people to know that if you do `---hostname test.com` you can't just then magically do `ssh test.com` and get your new VM.  But for these specific docs it's not a difference.  And I have frequently made VMs with hostnames and not bothered to set up any DNS and google will set the hostname on the machine without issue.